pkg/sentry/kernel/semaphore/semaphore.go

```diff
@@ -344,12 +290,32 @@ func (r *Registry) findFirstAvailableIndex() (int32, bool) {
 
 func (r *Registry) totalSems() int {
 	totalSems := 0
-	for _, v := range r.semaphores {
-		totalSems += v.Size()
+	for _, v := range r.reg.Objects() {
```

Comment:
You raised a concern about how the `ipc.Registry.Objects` method doesn't limit the caller to read-only access to the registry. You can enforce this by defining a new method on `ipc.Registry` instead of `ipc.Registry.Object`:

```
package ipc

func (r *Registry) ForAllObjects(f func(o *Object)) {
    for _, v := range r.objects {
        f(v)
    }
}
```

This should let you keep `ipc.reg.Objects` private and disallow external mutation of the map.

Reply:
Done